api/src/main/java/uk/gov/dhsc/htbhf/claimant/SchedulerConfig.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.dhsc.htbhf.claimant;
+
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.spi.JobFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.scheduling.quartz.SchedulerFactoryBean;
+import uk.gov.dhsc.htbhf.claimant.scheduler.AutowiringSpringBeanJobFactory;
+import uk.gov.dhsc.htbhf.claimant.scheduler.CreateCardJob;
+
+import javax.sql.DataSource;
+
+/**
+ * Configuration for scheduling the creation of new cards.
+ */
+@Configuration
+public class SchedulerConfig {
+
+    /**
+     * Create a job factory that is spring context aware.
+     * @param applicationContext spring application context.
+     * @return spring aware job factory
+     */
+    @Bean
+    public JobFactory jobFactory(ApplicationContext applicationContext) {
+        AutowiringSpringBeanJobFactory jobFactory = new AutowiringSpringBeanJobFactory();
+        jobFactory.setApplicationContext(applicationContext);
+        return jobFactory;
+    }
+
+    /**
+     * Creates a job detail instance containing the job to create new cards.
+     * @return job detail
+     */
+    @Bean
+    public JobDetail createCardJobDetail() {
+        return JobBuilder.newJob(CreateCardJob.class)
+                .withIdentity("Create new cards")
+                .storeDurably()
+                .build();
+    }
+
+    /**
+     * Trigger the job every hour and run indefinitely.
+     * @return the job trigger
+     */
+    @Bean
+    public Trigger createCardTrigger() {
+        SimpleScheduleBuilder scheduleBuilder = SimpleScheduleBuilder.simpleSchedule()
+                .withIntervalInHours(1).repeatForever();
+
+        return TriggerBuilder.newTrigger().forJob(createCardJobDetail())
```

Comment:
I presume this triggers the job immediately on application startup and then every hour after that, is that correct? Would be good to clarify when the first run happens as its not 100% clear.

Reply:
First run is one hour after start up, I've added javadoc to clarify this.

Reply:
Can we make the time interval configurable? Also, the javadoc says one hour but the code is every 10 seconds

Reply:
It did say an hour earlier, Dave seems to have changed it to every 10 seconds, I'm guessing for some local testing that you mistakenly committed Dave?

Reply:
Made configurable.